plugin/acl/setup_test.go

```diff
@@ -0,0 +1,294 @@
+package acl
+
+import (
+	"os"
+	"testing"
+
+	"github.com/caddyserver/caddy"
+)
+
+var (
+	setupTestFiles = map[string]string{
+		"acl-setup-test-1.txt": `10.218.128.0/24
+35.39.53.223/32
+43.105.127.35/18`,
+	}
+)
+
+func envSetup(files map[string]string) {
+	for k, v := range files {
+		file, err := os.Create(k)
+		defer file.Close()
+		if err != nil {
+			panic(err)
+		}
+		_, err = file.Write([]byte(v))
+		if err != nil {
+			panic(err)
+		}
+	}
+}
+
+func envCleanup(files map[string]string) {
+	for k := range files {
+		err := os.Remove(k)
+		if err != nil {
+			panic(err)
+		}
+	}
+}
+
+func Test_setup(t *testing.T) {
+	envSetup(setupTestFiles)
+	defer envCleanup(setupTestFiles)
+
+	tests := []struct {
+		name    string
+		ctr     *caddy.Controller
+		wantErr bool
+	}{
+		{
+			"Blacklist 1",
+			caddy.NewTestController("dns", `
+			acl {
+				block type A net 192.168.0.0/16
+			}
+			`),
+			false,
+		},
+		{
+			"Blacklist 2",
+			caddy.NewTestController("dns", `
+			acl {
+				block type ANY net 192.168.0.0/16
+			}
+			`),
+			false,
+		},
+		{
+			"Blacklist 3",
+			caddy.NewTestController("dns", `
+			acl {
+				block type A net ANY
+			}
+			`),
+			false,
+		},
+		{
+			"Blacklist 4",
+			caddy.NewTestController("dns", `
+			acl {
+				allow type ANY net 192.168.1.0/24
+				block type ANY net 192.168.0.0/16
+			}
+			`),
+			false,
+		},
+		{
+			"Whitelist 1",
+			caddy.NewTestController("dns", `
+			acl {
+				allow type ANY net 192.168.0.0/16
+				block type ANY net ANY
+			}
+			`),
+			false,
+		},
+		{
+			"fine-grained 1",
+			caddy.NewTestController("dns", `
+			acl a.example.org {
+				block type ANY net 192.168.1.0/24
+			}
+			`),
+			false,
+		},
+		{
+			"fine-grained 2",
+			caddy.NewTestController("dns", `
+			acl a.example.org {
+				block type ANY net 192.168.1.0/24
+			}
+			acl b.example.org {
+				block type ANY net 192.168.2.0/24
+			}
+			`),
+			false,
+		},
+		{
+			"multiple-networks 1",
+			caddy.NewTestController("dns", `
+			acl example.org {
+				block type ANY net 192.168.1.0/24 192.168.3.0/24
+			}
+			`),
+			false,
+		},
+		{
+			"multiple-networks 2",
+			caddy.NewTestController("dns", `
+			acl example.org {
+				block type ANY net 192.168.3.0/24
+			}
+			`),
+			false,
+		},
+		{
+			"Keyword PRIVATE 1",
+			caddy.NewTestController("dns", `
+			acl example.org {
+				block type ANY net PRIVATE
+			}
+			`),
+			false,
+		},
+		{
+			"Keyword LOCAL 1",
+			caddy.NewTestController("dns", `
+			acl example.org {
+				allow type ANY net LOCAL
+				block type ANY net ANY
+			}
+			`),
+			false,
+		},
+		{
+			"Local file 1",
+			caddy.NewTestController("dns", `
```

Comment:
you don't need to call caddy.NewTestController that often. please look at other tests.